### PR TITLE
feat(cloud-monitoring-exporter): Support Distribution exporting

### LINKED ISSUE #101: Add distribution and histogram metric support
Only non-distribution (bool, int64, double, etc.) values are supported by the exporter at the moment. See [this](https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/blob/a1f6264bec877bc00c66348d249ac076d8cf0207/packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts#L177) line of code for the relevant TODO and where support should be added (perhaps among other places).

### PR DESCRIPTION
**Pull request by markrzen**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue/feature](https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #101 
Replaces #409

### CODE REVIEW

File: packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts [hunk 1]
```diff
@@ -73,20 +76,27 @@ function transformMetricKind(kind: OTMetricKind): MetricKind {
...
-function transformValueType(valueType: OTValueType): ValueType {
+function transformValueType(
+  valueType: OTValueType,
+  metricKind: OTMetricKind
+): ValueType {
+  // Stackdriver ValueType is not a 1 to 1 with OTel ValueType
   if (valueType === OTValueType.DOUBLE) {
+    if (metricKind === OTMetricKind.HISTOGRAM) {
+      return ValueType.DISTRIBUTION;
+    }
```
Comment (markrzen) [inline packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts:99]: Histograms are both a OTValueType.DOUBLE and a OTMetricKind.HISTOGRAM.

File: packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts [hunk 2]
```diff
@@ -144,6 +157,7 @@ function transformPoint(
   switch (metricDescriptor.metricKind) {
     case OTMetricKind.COUNTER:
     case OTMetricKind.OBSERVABLE_COUNTER:
+    case OTMetricKind.HISTOGRAM:
```
Comment (markrzen) [inline packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts:160]: Histograms require a startDate.

**Comment by markrzen** [comment]
@dashpole OK, CLA signed. Comments copied over from the original PR #409, which got closed out when I sync'd back to my fork.

**Comment by markrzen** [comment]
@dashpole Are y'all trying to maintain support for no longer supported versions of Node? 10.x and 12.x?
If so, I will need to revert the mocha upgrades.

**Comment by markrzen** [comment]
@dashpole Went ahead and dropped mocha/ts-mocha back down to 9.x to allow for Node 10.x support.

**Comment by dashpole** [comment]
I'm not sure.  @aabmass should know.  I'm just here to poke the test runner. :)

**Comment by markrzen** [comment]
Tests passed. 👍 👍

**Comment by aabmass** [comment]
@markrzen thanks for the PR. It looks like this is still building against [@opentelemetry/sdk-metrics-base](https://www.npmjs.com/package/@opentelemetry/sdk-metrics-base) which is deprecated. I think the first step would be upgrading to a newer metrics release on the new packages since so much has changed upstream, and adapting the existing code. I think this would be a sizeable amount of work which i definitely don't eoxec

I apologize for not getting around to this work, I haven't had much time to maintain this work.

File: packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts [hunk 3]
```diff
@@ -73,20 +76,27 @@ function transformMetricKind(kind: OTMetricKind): MetricKind {
   switch (kind) {
     case OTMetricKind.COUNTER:
     case OTMetricKind.OBSERVABLE_COUNTER:
+    case OTMetricKind.HISTOGRAM:
```
[the remark this answers is not in the document]
Comment (aabmass) [reply packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts:79]: That's right. Btw, we've implemented some canonical mappings from OTel to GCM in the collector exporter. We hope to implement the same logic in the language exporters.

**Comment by markrzen** [comment]
Moved to using Prometheus for easier integration.

**Pull request closed without merging**